src/app/pages/checkout-payment/payment-concardis-creditcard-cvc-detail/payment-concardis-creditcard-cvc-detail.component.ts

```diff
@@ -0,0 +1,153 @@
+import { ChangeDetectionStrategy, ChangeDetectorRef, Component, Input, OnInit } from '@angular/core';
+import { takeUntil } from 'rxjs/operators';
+
+import { CheckoutFacade } from 'ish-core/facades/checkout.facade';
+import { PaymentInstrument } from 'ish-core/models/payment-instrument/payment-instrument.model';
+import { ScriptLoaderService } from 'ish-core/utils/script-loader/script-loader.service';
+
+import { PaymentConcardisComponent } from '../payment-concardis/payment-concardis.component';
+
+// allows access to concardis js functionality
+// tslint:disable-next-line:no-any
+declare var PayEngine: any;
+
+@Component({
+  selector: 'ish-payment-concardis-creditcard-cvc-detail',
+  templateUrl: './payment-concardis-creditcard-cvc-detail.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+// tslint:disable-next-line: rxjs-prefer-angular-takeuntil
+export class PaymentConcardisCreditcardCvcDetailComponent extends PaymentConcardisComponent implements OnInit {
+  @Input() paymentInstrument: PaymentInstrument;
+  validityTimeInMinutes: string;
+
+  constructor(
+    protected scriptLoader: ScriptLoaderService,
+    protected cd: ChangeDetectorRef,
+    private checkoutFacade: CheckoutFacade
+  ) {
+    super(scriptLoader, cd);
+  }
+
+  ngOnInit() {
+    super.formInit();
+  }
+  /**
+   * load concardis script if component is visible
+   */
+  loadScript() {
+    // load script only once if component becomes visible
+    if (!this.scriptLoaded) {
+      const merchantId = this.getParamValue(
+        'ConcardisPaymentService.MerchantID',
+        'checkout.credit_card.merchantId.error.notFound'
+      );
+      this.validityTimeInMinutes = this.getParamValue(
+        'intershop.payment.Concardis_CreditCard.cvcmaxage',
+        'checkout.credit_card.validityTime.error.notFound'
+      );
+      // if config params are missing - don't load script
+      if (!merchantId) {
+        return;
+      }
+
+      this.scriptLoaded = true;
+      this.scriptLoader
+        .load(this.getPayEngineURL())
+        .pipe(takeUntil(this.destroy$))
+        .subscribe(
+          () => {
+            PayEngine.setPublishableKey(merchantId);
+          },
+          error => {
+            this.scriptLoaded = false;
+            this.errorMessage.general.message = error;
+            this.cd.detectChanges();
+          }
+        );
+    }
+  }
+
+  isCvcExpired() {
+    // if cvc last updated timestamp is less than 25 minutes then return false
```

Comment:
where does the 25 come from? is it the validityTimeInMin?

Reply:
Done.